packages/grafana-data/src/types/fieldOverrides.ts

```diff
@@ -132,4 +132,5 @@ export enum FieldConfigProperty {
   Mappings = 'mappings',
   Links = 'links',
   Color = 'color',
+  ModeNullValue = 'nullValueMode',
```

Comment:
I'm not 100% sure we want this as a standard property but it feels like it might make sense. @ryantxu @torkelo, what's your opinion on that?

Reply:
yea, not sure. Maybe not something all panels get but still standard, and can be opted into? Tricky part is should it affect reduced values in legend etc? Or for that should we make something like this but a transformation instead? This is could also be done as a transformation, but is a common problem and feels cumbersome to have to add a transform for this

Reply:
@torkelo I was thinking about the possibility to have a transformer for this instead. But feels like it is so many steps (clicks) to get the expected behavior.

I don't have any strong opinions on this so point me in the right direction and I will apply the proper changes.

Reply:
So adding this as an std property has also a consequence of this null value mode being available for all React panels. Currently, we only have a way to explicitly say use all or only specified properties. There is no way to exclude some of the properties, which might be usefull in this case, so that 

I suggest having the null value mode as a custom property as it was before these changes.

Reply:
Interesting, nullValueMode is actually used in the fieldReducer, https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/transformations/fieldReducer.ts#L87  

But in many of the places it's used it's before field overrides has been applied so it wont have access to the final nullValueMode. hm..

Reply:
> Or for that should we make something like this but a transformation instead? This is could also be done as a transformation, but is a common problem and feels cumbersome to have to add a transform for this

from my very lacking understanding of transforms, how would this work?

if the graph has a single series, then you can probably use a transform to remove the nulls, which will work as "connected" or coerce to 0s for "as zeros" behavior, or leave as-is for "as gaps".

for graphs with multiple series, only the "as zeros" coercion and "as gaps" (do nothing) transforms will work, but "connected" cannot be attained by a remove-nulls transform as in the single series case, since all series must be x-aligned and the same length.

i think the only meaningful transform here that will work without caveats is nulls -> zeros?

Reply:
for uPlot, supporting "connected" mode will be super tricky.  that will take some thought, but I don't think can happen easily.

For reference, In plotly studio, they show options like this:
![image](https://user-images.githubusercontent.com/705951/96036660-a202dd00-0e19-11eb-8525-5618371b4e14.png)

This is their UI representation for:
https://plotly.com/javascript/reference/scattergl/#scattergl-line-shape

Reply:
> for uPlot, supporting "connected" mode will be super tricky

as long as alignment is maintained across all series (data frames are null-filled as needed), then it's already well-handled. the trickiness just migrates into the data alignment logic, no? 🤔

Reply:
But how would we solve connected in that case? Connecting one series would lead removing valid data points for another?

Reply:
does this help?

https://jsfiddle.net/qa9bsxvf/

Reply:
Just to be totally clear, I think connecting points work great out of the box in uPlot. We had that already working well until I have refactored the config to use geometries, and missed reimplementing it. 

Ref plotly studio and line rendering @ryantxu - I think series paths rendering which can be relatively easily modified in uPlot (ref https://leeoniya.github.io/uPlot/demos/line-stepped.html, https://leeoniya.github.io/uPlot/demos/line-smoothing.html)

Reply:
@dprokop 

as a bonus, https://github.com/leeoniya/uPlot/commit/1c634ea55312b132addd7f3674fd8f03305080cc exposes `self.paths` which makes it possible to dynamically re-use the internal path builder. this can be useful if you want to render stepped paths when zoomed in, but fall back to the default line path builder at lower zoom levels when the higher data density makes stepped rendering moot.

the stepped example now includes a demo of how to do this: https://leeoniya.github.io/uPlot/demos/line-stepped.html